doc-gen/src/commands/generate.ts

```diff
@@ -135,3 +120,83 @@ async function getPackageJson(
 
   return JSON.parse(fs.readFileSync(path, "utf-8"));
 }
+
+function generateFunctionDocs(
+  node: ts.VariableStatement | ts.FunctionDeclaration | ts.ClassElement,
+  importedName: string,
+  outputFilePath: string,
+  packageName: string,
+  type: "tsx" | "ts" | "classMethod",
+  importOverride?: string
+) {
+  const jsdocCommentAndTags = ts.getJSDocCommentsAndTags(node);
+  if (jsdocCommentAndTags.length === 0) return;
+
+  const documentation = functionTemplate(
+    importedName,
+    packageName,
+    jsdocCommentAndTags,
+    importOverride
+  );
+
+  // TODO: need to handle this differently in case we have `use*` methods that aren't hooks
+  const outputLocation =
+    type === "classMethod"
+      ? ""
+      : importedName.startsWith("use")
+      ? "./hooks"
+      : type === "tsx"
+      ? "./components"
+      : "./functions";
+
+  fs.outputFileSync(
+    path.resolve(outputFilePath, outputLocation, `${importedName}.mdx`),
+    // I have 0 clue why this needs to be formatted twice to get the correct output, but here we are...
+    format(format(documentation, { parser: "mdx" }), { parser: "mdx" })
+  );
+}
+
+function generateClassDocs(
+  node: ts.ClassDeclaration,
+  outputFilePath: string,
+  importedName: string,
+  packageName: string
+) {
+  const classOutputBasePath = path.resolve(
+    outputFilePath,
+    `./classes/${importedName}`
+  );
+
+  node.members.forEach((member) => {
+    if (
+      ts.isPropertyDeclaration(member) &&
+      (!member.initializer ||
+        !ts.isArrowFunction(member.initializer) ||
+        !ts.isFunctionExpression(member.initializer))
+    ) {
+      // skip properties that aren't public functions
+      return;
+    }
+
+    if (
+      ts.isConstructorDeclaration(member) ||
+      ts.isMethodDeclaration(member) ||
+      (ts.isPropertyDeclaration(member) &&
+        member.initializer &&
+        (ts.isArrowFunction(member.initializer) ||
+          ts.isFunctionExpression(member.initializer)))
+    ) {
+      // TODO: need to handle highlighting that a class extends another class. This should result in mention in the constructor docs
```

Comment:
also need to handle generating documentation for the inherited classes 🤔 might do this by just exporting the base classes for now and doing this TODO rather than generating docs for those methods by traversing the tree. Traversing the tree is tricky because the examples wouldn't be for the specific class but might a more generic example